### PR TITLE
docs(cmp): Document ConfigManagementPlugin.spec.version more clearly

### PR DESCRIPTION
**Pull request by peschmae**
Adds a description to the `spec.version` attribute in the CMP, to make it clearer that this is the version of the plugin, and is optional.

This confused me quite a bit in the begining, and there were already tickets about this, eg. #13034 

Checklist:

* [x ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x ] The title of the PR states what changed and the related issues number (used for the release note).
* [ x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ x] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

**Comment by peschmae** [comment]
Rebased to master

File: docs/operator-manual/config-management-plugins.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ metadata:
   # The name of the plugin must be unique within a given Argo CD instance.
   name: my-plugin
 spec:
+  # The version of your plugin. Optional, will be appended to <metadata.name> if set.
```
Comment (crenshaw-dev) [inline docs/operator-manual/config-management-plugins.md:37]:
```suggestion
  # The version of your plugin. Optional. If specified, the Application's spec.source.plugin.name field
  # must be <plugin name>-<plugin version>.
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Maybe just a bit more explicit?

File: docs/operator-manual/config-management-plugins.md (hunk 1 above)
Comment (peschmae) [reply docs/operator-manual/config-management-plugins.md]: I've adpoted your wording, and ammended the commit.

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**